### PR TITLE
Fix#AC-2494 No indication of required input fields (pattern: Required…

### PR DESCRIPTION
**Pull request by RaghavendraTirumalasetti**
… fields)

## Description

Reproduction Steps

Locations (representative sample):

Shopping Bag
Checkout - Shipping Information
Checkout - Payment
- Checkout - Confirmation / Create an Account
Sign In

1. Visually inspect form elements

Actual Behavior

There is no indication of which form fields are required. Examples include:

Module 05b - Shopping Bag
Module 06a - Checkout - Shipping Information
Module 06b - Checkout - Payment
Module 06c - Checkout - Confirmation / Create an Account
Module 07 - Sign In

Expected Behavior

Provide visual and programmatic instructions to indicate which fields are required.

## Related Issue

Closes https://jira.corp.adobe.com/browse/AC-2494

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### Resolved issues:
1. [x] resolves magento/pwa-studio#3991: Fix#AC-2494 No indication of required input fields (pattern: Required…

### CODE REVIEW

**Comment by anthoula** [comment]
@magento create issue from PR

**Review by anthoula — CHANGES_REQUESTED** [review]
Thanks @RaghavendraTirumalasetti - Can you please add translations to all new strings?

File: packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js [hunk 1]
```diff
@@ -99,6 +99,10 @@ const CreateAccount = props => {
                     <TextInput
                         field="customer.firstname"
                         autoComplete="given-name"
+                        aria-label={formatMessage({
+                            id: 'Form.fistname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js:103]:
```suggestion
                            id: 'global.firstNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js [hunk 2]
```diff
@@ -113,6 +117,10 @@ const CreateAccount = props => {
                     <TextInput
                         field="customer.lastname"
                         autoComplete="family-name"
+                        aria-label={formatMessage({
+                            id: 'Form.lastname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js:121]:
```suggestion
                            id: 'global.lastNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js [hunk 3]
```diff
@@ -127,6 +135,10 @@ const CreateAccount = props => {
                     <TextInput
                         field="customer.email"
                         autoComplete="email"
+                        aria-label={formatMessage({
+                            id: 'Form.Email',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js:139]:
```suggestion
                            id: 'global.emailRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js [hunk 4]
```diff
@@ -147,6 +159,10 @@ const CreateAccount = props => {
                         validatePassword
                     ])}
                     validateOnBlur
+                    aria-label={formatMessage({
+                        id: 'Form.Password',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/OrderConfirmationPage/createAccount.js:163]:
```suggestion
                        id: 'global.passwordRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 5]
```diff
@@ -159,6 +159,10 @@ const CustomerForm = props => {
                             id="customer_firstname"
                             data-cy="CustomerForm-firstName"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.fistname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:163]:
```suggestion
                                id: 'global.firstNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 6]
```diff
@@ -176,13 +180,21 @@ const CustomerForm = props => {
                             id="customer_lastname"
                             data-cy="CustomerForm-lastName"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.lastname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:184]:
```suggestion
                                id: 'global.lastNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 7]
```diff
@@ -176,13 +180,21 @@ const CustomerForm = props => {
...
+                            })}
                         />
                     </Field>
                 </div>
                 <div className={classes.country}>
                     <Country
                         validate={isRequired}
                         data-cy="CustomerForm-country"
+                        aria-label={formatMessage({
+                            id: 'Form.country',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:195]:
```suggestion
                            id: 'global.countryRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 8]
```diff
@@ -198,6 +210,10 @@ const CustomerForm = props => {
                             validate={isRequired}
                             id="customer_street0"
                             data-cy="CustomerForm-street0"
+                            aria-label={formatMessage({
+                                id: 'Form.streetAddress',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:214]:
```suggestion
                                id: 'global.streetAddressRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 9]
```diff
@@ -224,12 +240,20 @@ const CustomerForm = props => {
                             id: 'global.city',
                             defaultMessage: 'City'
                         })}
+                        aria-label={formatMessage({
+                            id: 'Form.country',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:244]:
```suggestion
                            id: 'global.countryRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 10]
```diff
@@ -224,12 +240,20 @@ const CustomerForm = props => {
...
+                            defaultMessage: 'Country Required'
+                        })}
                     >
                         <TextInput
                             field="city"
                             validate={isRequired}
                             id="customer_city"
                             data-cy="CustomerForm-city"
+                            aria-label={formatMessage({
+                                id: 'Form.city',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:254]:
```suggestion
                                id: 'global.cityRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 11]
```diff
@@ -240,12 +264,20 @@ const CustomerForm = props => {
                         fieldInput={'region[region]'}
                         fieldSelect={'region[region_id]'}
                         optionValueKey={'id'}
+                        aria-label={formatMessage({
+                            id: 'Form.state',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:268]:
```suggestion
                            id: 'global.stateRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 12]
```diff
@@ -240,12 +264,20 @@ const CustomerForm = props => {
...
+                            defaultMessage: 'State Required'
+                        })}
                     />
                 </div>
                 <div className={classes.postcode}>
                     <Postcode
                         validate={isRequired}
                         data-cy="CustomerForm-postcode"
+                        aria-label={formatMessage({
+                            id: 'Form.postalCode',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:278]:
```suggestion
                            id: 'global.postalCodeRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js [hunk 13]
```diff
@@ -261,6 +293,10 @@ const CustomerForm = props => {
                             validate={isRequired}
                             id="customer_telephone"
                             data-cy="CustomerForm-telephone"
+                            aria-label={formatMessage({
+                                id: 'Form.phonenumber',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/customerForm.js:297]:
```suggestion
                                id: 'global.phoneNumberRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 14]
```diff
@@ -153,6 +153,10 @@ const GuestForm = props => {
                             id="email"
                             data-cy="GuestForm-email"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.Email',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:157]:
```suggestion
                                id: 'global.emailRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 15]
```diff
@@ -185,6 +189,10 @@ const GuestForm = props => {
                             id="firstname"
                             data-cy="GuestForm-firstName"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.fistname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:193]:
```suggestion
                                id: 'global.firstNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 16]
```diff
@@ -205,6 +213,10 @@ const GuestForm = props => {
                             id="lastname"
                             data-cy="GuestForm-lastName"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.lastname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:217]:
```suggestion
                                id: 'global.lastNameRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 17]
```diff
@@ -216,6 +228,10 @@ const GuestForm = props => {
                         })}
                         validate={isRequired}
                         data-cy="GuestForm-country"
+                        aria-label={formatMessage({
+                            id: 'Form.country',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:232]:
```suggestion
                            id: 'global.countryRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 18]
```diff
@@ -235,6 +251,10 @@ const GuestForm = props => {
                             id="street0"
                             data-cy="GuestForm-street0"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.streetAddress',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:255]:
```suggestion
                                id: 'global.streetAddressRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 19]
```diff
@@ -275,6 +295,10 @@ const GuestForm = props => {
                             id="city"
                             data-cy="GuestForm-city"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.city',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:299]:
```suggestion
                                id: 'global.cityRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 20]
```diff
@@ -289,6 +313,10 @@ const GuestForm = props => {
                         fieldSelect={'region[region_id]'}
                         optionValueKey={'id'}
                         data-cy="GuestForm-region"
+                        aria-label={formatMessage({
+                            id: 'Form.state',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:317]:
```suggestion
                            id: 'global.stateRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 21]
```diff
@@ -299,6 +327,10 @@ const GuestForm = props => {
                         })}
                         validate={isRequired}
                         data-cy="GuestForm-postcode"
+                        aria-label={formatMessage({
+                            id: 'Form.postalCode',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:331]:
```suggestion
                            id: 'global.postalCodeRequired',
```

File: packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js [hunk 22]
```diff
@@ -318,6 +350,10 @@ const GuestForm = props => {
                             id="telephone"
                             data-cy="GuestForm-telephone"
                             validate={isRequired}
+                            aria-label={formatMessage({
+                                id: 'Form.phonenumber',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CheckoutPage/ShippingInformation/AddressForm/guestForm.js:354]:
```suggestion
                                id: 'global.phoneNumberRequired',
```

File: packages/venia-ui/lib/components/CreateAccount/createAccount.js [hunk 23]
```diff
@@ -101,6 +101,10 @@ const CreateAccount = props => {
                     mask={value => value && value.trim()}
                     maskOnBlur={true}
                     data-cy="customer-firstname"
+                    aria-label={formatMessage({
+                        id: 'Form.fistname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CreateAccount/createAccount.js:105]:
```suggestion
                        id: 'global.firstNameRequired',
```

File: packages/venia-ui/lib/components/CreateAccount/createAccount.js [hunk 24]
```diff
@@ -119,6 +123,10 @@ const CreateAccount = props => {
                     mask={value => value && value.trim()}
                     maskOnBlur={true}
                     data-cy="customer-lastname"
+                    aria-label={formatMessage({
+                        id: 'Form.lastname',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CreateAccount/createAccount.js:127]:
```suggestion
                        id: 'global.lastNameRequired',
```

File: packages/venia-ui/lib/components/CreateAccount/createAccount.js [hunk 25]
```diff
@@ -137,6 +145,10 @@ const CreateAccount = props => {
                     mask={value => value && value.trim()}
                     maskOnBlur={true}
                     data-cy="customer-email"
+                    aria-label={formatMessage({
+                        id: 'Form.Email',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CreateAccount/createAccount.js:149]:
```suggestion
                        id: 'global.emailRequired',
```

File: packages/venia-ui/lib/components/CreateAccount/createAccount.js [hunk 26]
```diff
@@ -157,6 +169,10 @@ const CreateAccount = props => {
                 mask={value => value && value.trim()}
                 maskOnBlur={true}
                 data-cy="password"
+                aria-label={formatMessage({
+                    id: 'Form.Password',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/CreateAccount/createAccount.js:173]:
```suggestion
                    id: 'global.passwordRequired',
```

File: packages/venia-ui/lib/components/SignIn/signIn.js [hunk 27]
```diff
@@ -78,6 +78,10 @@ const SignIn = props => {
                         field="email"
                         validate={isRequired}
                         data-cy="email"
+                        aria-label={formatMessage({
+                            id: 'Form.Email',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/SignIn/signIn.js:82]:
```suggestion
                            id: 'global.emailRequired',
```

File: packages/venia-ui/lib/components/SignIn/signIn.js [hunk 28]
```diff
@@ -92,6 +96,10 @@ const SignIn = props => {
                     autoComplete="current-password"
                     isToggleButtonHidden={false}
                     data-cy="password"
+                    aria-label={formatMessage({
+                        id: 'Form.Password',
```
Comment (anthoula) [inline packages/venia-ui/lib/components/SignIn/signIn.js:100]:
```suggestion
                        id: 'global.passwordRequired',
```

**Review by anthoula — CHANGES_REQUESTED** [review]
Thanks @RaghavendraTirumalasetti ! Most of these can be global translation IDs.

**Review by anthoula — APPROVED** [review]

**Comment by dpatil-magento** [comment]
run cypress

**Pull request merged**